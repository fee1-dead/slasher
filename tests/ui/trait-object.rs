#![deny(redetect::unused)]

// make sure that trait object usage is marked as used

trait MyTrait {}

impl MyTrait for u32 {}
impl MyTrait for u8 {}
//~^ ERROR: implementation of MyTrait for u8 is unused

fn u(_x: &dyn MyTrait) {}

fn main() {
    u(&1u32);
}

