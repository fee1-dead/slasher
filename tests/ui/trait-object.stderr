error: implementation of MyTrait for u8 is unused
 --> tests/ui/trait-object.rs:8:1
  |
8 | impl MyTrait for u8 {}
  | ^^^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> tests/ui/trait-object.rs:1:9
  |
1 | #![deny(redetect::unused)]
  |         ^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

